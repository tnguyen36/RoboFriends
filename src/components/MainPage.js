import React from 'react';
import CardList from '../components/CardList';
import Scroll from '../components/Scroll';
import SearchBox from '../components/SearchBox';
import ErrorBoundry from '../components/ErrorBoundry';
import Header from '../components/Header';




class MainPage extends React.Component {
    

    componentDidMount() {
        this.props.requestRobots();
        
    }

    onSearchChange = (event) => {
        this.props.setSearchField(event.target.value);
    }

    filteredRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        });
    }
    
    render() {
        const {isPending} = this.props;       
        return isPending ?
            <h1>Loading</h1> :      
        (
            <div className='tc'>
                <Header />
                <SearchBox searchChange={this.onSearchChange} />
                <Scroll>
                    <ErrorBoundry>
                        <CardList robots={this.filteredRobots()} />
                    </ErrorBoundry>
                </Scroll>
                
            </div>
        );
        
    }
}



export default MainPage;
