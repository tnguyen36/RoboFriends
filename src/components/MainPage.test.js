import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import React from 'react';
import MainPage from './MainPage';

let wrapper;
let mockProps;

beforeEach(() => {
    mockProps = {
        requestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps} />)
    
})

it('renders MainPage without crashing', () => {
    const tree = renderer.create(<MainPage {...mockProps} />).toJSON()
    expect(tree).toMatchSnapshot()
})

it('filters robots correctly', () => {
    const mockProps2 = {
        requestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: 'john',
        isPending: false
    }
    const wrapper2 = shallow(<MainPage {...mockProps2} />)
    expect(wrapper2.instance().filteredRobots()).toEqual([{       
        id: 3,
        name: 'John',
        email: 'john@gmail.com'
    }])
})

it('filters robots correctly 2', () => {
    const mockProps3 = {
        requestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: 'a',
        isPending: false
    }
    const filteredRobots = []
    const wrapper3 = shallow(<MainPage {...mockProps3} />)
    expect(wrapper3.instance().filteredRobots()).toEqual(filteredRobots);
})

