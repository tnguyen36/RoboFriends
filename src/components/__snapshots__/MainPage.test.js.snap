// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <div>
    <h1
      className="f2"
    >
      RoboFriends
    </h1>
    <button
      color="red"
      id="counter"
      onClick={[Function]}
    >
      Count: 
      0
    </button>
  </div>
  <div
    className="pa2"
  >
    <input
      aria-label="Search Robots"
      className="pa3 ba b--green bg-lightest-blue"
      onChange={[Function]}
      placeholder="search robots"
      type="search"
    />
  </div>
  <div
    style={
      Object {
        "border": "1px solid black",
        "height": "800px",
        "overflowY": "scroll",
      }
    }
  >
    <div />
  </div>
</div>
`;
